pkg/operator/controllers/storageaccounts/storageaccounts.go

```diff
@@ -0,0 +1,82 @@
+package storageaccounts
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"strings"
+
+	mgmtstorage "github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2019-06-01/storage"
+	"github.com/Azure/go-autorest/autorest/to"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+func (r *reconcileManager) reconcileAccounts(ctx context.Context) error {
+	resourceGroup := stringutils.LastTokenByte(r.instance.Spec.ClusterResourceGroupID, '/')
+
+	subnets, err := r.kubeSubnets.List(ctx)
+	if err != nil {
+		return err
+	}
+
+	rc, err := r.imageregistrycli.ImageregistryV1().Configs().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	storageAccounts := []string{
+		"cluster" + r.instance.Spec.StorageSuffix, // this is our creation, so name is deterministic
+		rc.Spec.Storage.Azure.AccountName,
+	}
+
+	for _, accountName := range storageAccounts {
+		var changed bool
+
+		account, err := r.storage.GetProperties(ctx, resourceGroup, accountName, "")
+		if err != nil {
+			return err
+		}
+
+		for _, subnet := range subnets {
```

Comment:
I'm pretty sure we're missing the gateway and RP subnets here.  The gateway subnet needs to be allowed on everything, but the RP only needs access to the cluster storage account which contains the original ignition config read out during adminUpdates.  

We may need to export these into the cluster CRD and ensure that it is added for egress lockdown support for existing clusters.

Reply:
We may want to coordinate with @SudoBrendan  since he has a full service RP up and we can test with the gateway in place.

Reply:
Addressed in https://github.com/Azure/ARO-RP/pull/1865/commits/f8257e6b0e33fdb91cfcd8b314a892eedeb2afab